include/wil/com.h

```diff
@@ -3002,6 +3002,150 @@ namespace wil
 
 #endif // __IObjectWithSite_INTERFACE_DEFINED__
 
+
+// if C++17 or greater
+#if WIL_HAS_CXX_17
+#ifdef WIL_ENABLE_EXCEPTIONS
+namespace details
+{
+    template<typename>
+    struct com_enumerator_next_traits;
+
+    template<typename Itf, typename T, typename Ret>
+    struct com_enumerator_next_traits<Ret(__stdcall Itf::*)(ULONG, T*, ULONG*)>
+    {
+        using Interface = Itf;
+        using Result = T;
+    };
+
+    template<typename Itf, typename T, typename Ret>
+    struct com_enumerator_next_traits<Ret(__stdcall Itf::*)(ULONG, T*, ULONG*) noexcept>
+    {
+        using Interface = Itf;
+        using Result = T;
+    };
+
+    template<typename T>
+    struct has_next
+    {
+      template <typename U = T>
+      static auto test(int) -> decltype(wistd::declval<U>()->Next(0, nullptr, nullptr), wistd::true_type{});
+
+      template <typename>
+			static auto test(...) -> wistd::false_type;
+
+			static constexpr bool value = decltype(test<T>(0))::value;
+    };

```

Comment:
```suggestion
    template<typename T>
    struct has_next
    {
        template <typename U = T>
        static auto test(int) -> decltype(wistd::declval<U>()->Next(0, nullptr, nullptr), wistd::true_type{});

        template <typename>
        static auto test(...) -> wistd::false_type;

        static constexpr bool value = decltype(test<T>(0))::value;
    };
```
Something wonky happened with the formatting here

Reply:
oh weird. I think that was probably ghcp using tabs instead of spaces :(